instrumentation/opentelemetry-instrumentation-httpx/tests/test_httpx_integration.py

```diff
@@ -1214,3 +1226,36 @@ def test_basic_multiple(self):
         self.perform_request(self.URL, client=self.client)
         self.perform_request(self.URL, client=self.client2)
         self.assert_span(num_spans=2)
+
+    def test_async_response_hook_does_nothing_if_not_coroutine(self):
+        HTTPXClientInstrumentor().instrument(
+            tracer_provider=self.tracer_provider,
+            async_response_hook=_response_hook,
+        )
+        client = self.create_client()
+        result = self.perform_request(self.URL, client=client)
+
+        self.assertEqual(result.text, "Hello!")
+        span = self.assert_span()
+        self.assertEqual(
+            dict(span.attributes),
+            {
+                SpanAttributes.HTTP_METHOD: "GET",
+                SpanAttributes.HTTP_URL: self.URL,
+                SpanAttributes.HTTP_STATUS_CODE: 200,
+            },
+        )
+        HTTPXClientInstrumentor().uninstrument()
+
+    def test_async_request_hook_does_nothing_if_not_coroutine(self):
+        HTTPXClientInstrumentor().instrument(
+            tracer_provider=self.tracer_provider,
+            async_request_hook=_request_hook,
+        )
+        client = self.create_client()
+        result = self.perform_request(self.URL, client=client)
+
+        self.assertEqual(result.text, "Hello!")
+        span = self.assert_span()
+        self.assertEqual(span.name, "GET")
+        HTTPXClientInstrumentor().uninstrument()
```

Comment:
Are we missing out on testing the issue behavior? In cases we have only request_hook, it shouldn't generate an error anymore:

For example, this should fail without your fix:
```
            HTTPXClientInstrumentor().instrument(
                tracer_provider=self.tracer_provider,
                request_hook=_request_hook,
            )
            client = self.create_client()
            result = self.perform_request(self.URL, client=client)

            self.assertEqual(result.text, "Hello!")
            span = self.assert_span()
            self.assertEqual(span.name, "GET" + self.URL)
            HTTPXClientInstrumentor().uninstrument()
```
Will produce the error: 

```
        with self._tracer.start_as_current_span(
            span_name, kind=SpanKind.CLIENT, attributes=span_attributes
        ) as span:
            exception = None
            if callable(self._request_hook):
>               await self._request_hook(span, request_info)
E               TypeError: object NoneType can't be used in 'await' expression
```

Reply:
We can be more defensive but I think the regression was to use a sync function in the async path under the hood. Given that we have two different parameters for sync and async hooks using a sync function with an async client I think is a progamming error more than an issue of the instrumentation.

Reply:
Maybe we can add a check when using it too

Reply:
Right. Taking a second look at the tests I'm fine with